spec-main/api-safe-storage-spec.ts

```diff
@@ -0,0 +1,83 @@
+import * as cp from 'child_process';
+import * as path from 'path';
+import { app, safeStorage } from 'electron/main';
+import { expect } from 'chai';
+import { emittedOnce } from './events-helpers';
+
+describe('safeStorage module', () => {
+  afterEach(() => {
+    app.quit();
+  });
+
+  describe('SafeStorage.isEncryptionAvailable()', () => {
+    it('should return true when encryption key is available', () => {
+      expect(safeStorage.isEncryptionAvailable()).to.equal(true);
+    });
+  });
+
+  describe('SafeStorage.encryptString()', () => {
+    it('valid input should correctly encrypt string', () => {
+      const plaintext = 'plaintext';
+      const encrypted = safeStorage.encryptString(plaintext);
+      expect(Buffer.isBuffer(encrypted)).to.equal(true);
+    });
+
+    it('UTF-16 characters can be encrypted', () => {
+      const plaintext = '€ - utf symbol';
+      const encrypted = safeStorage.encryptString(plaintext);
+      expect(Buffer.isBuffer(encrypted)).to.equal(true);
+    });
+  });
+
+  describe('SafeStorage.decryptString()', () => {
+    it('valid input should correctly decrypt string', () => {
+      const encrypted = safeStorage.encryptString('plaintext');
+      expect(safeStorage.decryptString(encrypted)).to.equal('plaintext');
+    });
+
+    it('UTF-16 characters can be decrypted', () => {
+      const plaintext = '€ - utf symbol';
+      const encrypted = safeStorage.encryptString(plaintext);
+      expect(safeStorage.decryptString(encrypted)).to.equal(plaintext);
+    });
+
+    it('unencrypted input should not throw', () => {
+      const plaintextBuffer = Buffer.from('I am unencoded!', 'utf-8');
+      expect(() => {
+        safeStorage.decryptString(plaintextBuffer);
```

Comment:
what _does_ this return...?

Reply:
returns 'I am unencoded!' - I will reflect this in the test

Reply:
hm... that's a bit surprising! i would expect that if i passed invalid input to this function, it would throw an error.

Reply:
I can look into this! Wondering what your thoughts on: if the chromium API that decrypts returns the plaintext string by design would it make sense to manually error out our electron API instead? 

Eg. We pass in "plaintext" to be decrypted and we get "plaintext" back- thus the electron api returns an error

Reply:
Hm, I don't think we should go so far as to check the output of the function, but I'm surprised that passing in `X` and receiving `X` in return happens without the underlying API throwing an error, and it seems like the sort of thing that might indicate incorrect error handling happening somewhere.

Reply:
i.e. if we're 100% sure that the underlying OSCrypt functions behave this way without indicating any error, then I think it's fine to pass this along & we can document it.

Reply:
From what I understand from this code within `OSCrypt` behaves this way assuming that if there is no prefix on the encrypted string, it treats it as plaintext and throws it back at us- I'll just document this within the`.md` file to avoid confusion

![image](https://user-images.githubusercontent.com/33054982/125526197-a491b6e3-53ca-4655-80ab-f1a1fb8bc155.png)

Reply:
`electron/src/components/os_crypt/os_crypt.mm`

Reply:
Ah, great, thanks for the verification. 👍